java/security/audit/cbc-padding-oracle.yaml

```diff
@@ -0,0 +1,18 @@
+rules:
+- id: cbc-padding-oracle
+  message: |
+    Using CBC with PKCS5Padding is susceptible to padding orcale attacks. A malicious actor
+    could discern the difference between plaintext with valid or invalid padding. Further,
+    CBC mode does not include any integrity checks. See https://find-sec-bugs.github.io/bugs.htm#CIPHER_INTEGRITY.
+    Use 'AES/GCM/NoPadding' instead.
+  metadata:
+    cwe: 'CWE-696: Incorrect Behavior Order'
+    owasp: 'A3: Sensitive Data Exposure'
+    source-rule-url: https://find-sec-bugs.github.io/bugs.htm#PADDING_ORACLE
+    references:
+    - https://capec.mitre.org/data/definitions/463.html
+  severity: WARNING
+  pattern: $CIPHER.getInstance("AES/CBC/PKCS5Padding");
```

Comment:
I feel you should pull this to $MODE variable and do python-where to make it broadly applicable.

Reply:
Oh, yeah, good call. Didn't even cross my mind that CBC could be with other algo's. 🤦‍♂️